Makefile

```
@@ -0,0 +1,30 @@
+.PHONY: clean build lint docs
+
+clean: # clean all build, python, and testing files
+	rm -fr build/
+	rm -fr dist/
+	rm -fr .eggs/
+	find . -name '*.egg-info' -exec rm -fr {} +
+	find . -name '*.egg' -exec rm -f {} +
+	find . -name '*.pyc' -exec rm -f {} +
+	find . -name '*.pyo' -exec rm -f {} +
+	find . -name '*~' -exec rm -f {} +
+	find . -name '__pycache__' -exec rm -fr {} +
+	rm -fr .tox/
+	rm -fr .coverage
+	rm -fr coverage.xml
+	rm -fr htmlcov/
+	rm -fr .pytest_cache
+	rm -fr .mypy_cache
+
+build: # run tests
+	pytest
+
+lint: # run formatting and linting
+	black -l 88 .
```

This checks the entire top-level directory by default which also includes `.conda` and other python files.

We can restrict the check to only places where we expect to have python scripts.

```suggestion
	black -l 88 ./subcell_analysis
    black -l 88 ./notebooks
```

Alternatively, we can move the `notebooks` directory into `subcell_analysis` to have a single entry point

---

Updated to restrict the linting to the two folders. I think it does make sense to move `notebooks` under `subcell_analysis` but we should do that in a separate PR.